app/services/courseService.js

```diff
@@ -77,8 +84,8 @@ angular
         }
 
         function addData(course) {
-            course.id = data[data.length - 1] + 1;
-            course.date = new Date().toLocaleDateString();
+            course.id = data.length;
```

Comment:
Looks like we can get situation when data will have a few equal ids.

Reply:
fix